File: Assessment_20_Questions/src/test/java/com/pages/SearchPage.java
package com.pages;

import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class SearchPage {

    WebDriver driver;
    public SearchPage(WebDriver driver) {
        this.driver = driver;
        PageFactory.initElements(driver, this);
    }

    @FindBy(name = "search")
    WebElement searchbtn;

    public void searchProduct(String product) {
        searchbtn.sendKeys(product, Keys.ENTER);
    }
}

File: Akksheetha M - Assessment 24/ECommerceAutomation/Question2/pages/search_page.py
from selenium.webdriver.common.by import By

class SearchPage:
    def __init__(self, driver):
        self.driver = driver

    def search_product(self, keyword):
        self.driver.find_element(By.NAME,"search").send_keys(keyword)
        self.driver.find_element(By.XPATH,"//button[@class='btn btn-default btn-lg']").click()
        results = self.driver.find_elements(By.XPATH,"//div[@class='caption']/h4/a")
        return [result.text for result in results]